expression/builtin_miscellaneous_vec.go

```diff
@@ -281,11 +283,88 @@ func (b *builtinNameConstTimeSig) vecEvalTime(input *chunk.Chunk, result *chunk.
 }
 
 func (b *builtinSleepSig) vectorized() bool {
-	return false
+	return true
 }
 
+// vecEvalInt evals a builtinSleepSig in a vectorized manner.
+// See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_sleep
 func (b *builtinSleepSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+
+	err = b.args[0].VecEvalReal(b.ctx, input, buf)
+	if err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+
+	for i := 0; i < n; i++ {
+		isNull := buf.IsNull(i)
+		val := buf.GetFloat64(i)
+
+		sessVars := b.ctx.GetSessionVars()
+		if isNull {
+			if sessVars.StrictSQLMode {
+				err := errIncorrectArgs.GenWithStackByArgs("sleep")
+				sessVars.StmtCtx.AppendError(err)
```

Comment:
```suggestion
```
Although it's a problem for TiDB which does not `AppendError`, actually I prefer handling `AppendError` commonplace but not every place error returned.